phoenix-core/src/main/java/org/apache/phoenix/compile/ExpressionCompiler.java

```diff
@@ -523,7 +523,12 @@ public Expression visitLeave(LikeParseNode node, List<Expression> children) thro
                 byte[] wildcard = {StringUtil.MULTI_CHAR_LIKE};
                 StringUtil.fill(nullExpressionString, 0, pattern.length(), wildcard, 0, 1, false);
                 if (pattern.equals(new String (nullExpressionString))) {
-                    return IsNullExpression.create(lhs, true, context.getTempPtr());
+                    if (node.isNegate()) {
+                        return LiteralExpression.newConstant(false, Determinism.ALWAYS);
```

Comment:
So WHERE col IS NOT '%' always returns false? Or should it become WHERE col IS NULL?